facebook/detectors/rce/cisco_smi/src/main/java/com/google/tsunami/plugins/detectors/rce/cisco_smi/CiscoSMIDetector.java

```diff
@@ -0,0 +1,135 @@
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.ForServiceName;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+import java.net.Socket;
+import java.util.Arrays;
+import javax.net.SocketFactory;
+
+
+/** A {@link VulnDetector} that detects vulnerable Cisco Smart Install protocol */
+@ForServiceName({"smart-install"})
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Cisco_SMI",
+    version = "0.1",
+    description =
+        "This detector checks for Cisco Smart Install client protocol.",
+    author = "Adrien Schildknecht (vulnscan@fb.com)",
+    bootstrapModule = CiscoSMIDetectorBootstrapModule.class)
+public final class CiscoSMIDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final SocketFactory socketFactory;
+
+  @Inject
+  public CiscoSMIDetector(@UtcClock Clock utcClock, SocketFactory socketFactory) {
+    this.utcClock = checkNotNull(utcClock);
+    this.socketFactory = checkNotNull(socketFactory);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    byte[] actualResponse = new byte[24];
+    // See https://github.com/Cisco-Talos/smi_check for more info about the payload
+    byte[] request = {
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    };
+    byte[] expectedResponse = {
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    };
+    HostAndPort hp = NetworkEndpointUtils.toHostAndPort(networkService.getNetworkEndpoint());
+
+    try {
+        Socket socket = socketFactory.createSocket(hp.getHost(), hp.getPort());
```

Comment:
code style (Google Java Style [section 4.2](https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation)): "Each time a new block or block-like construct is opened, the indent increases by two spaces". Can you please be consistent in the PR and use +2 spaces for indentation?